### ISSUE TITLE
ITestListener's methods get called multiple times for one test, when @Listeners annotation is used in multiple test classes

### ISSUE THREAD

**Issue by gib100** [cbeust/testng#2220]
This is not the biggest issue, but I guess reporting is fun. :)

### TestNG Version
7.1.0

### Expected behavior
When running from testng.xml, and specifying the listener class using the `@Listeners` annotation at test class level (in multiple test classes), ITestListener's methods should only be executed once per `<test>` defined in testng.xml.

### Actual behavior
When running from testng.xml, ITestListener's methods are executed twice for the first `<test>`, and once for the second `<test>` defined in testng.xml.

If there are three tests defined in testng.xml, then the listener methods are called three times for the first `<test>`, two times for the second `<test>`, and once for the third `<test>`.

- When running only a specific test class (not running from testng.xml), there is no problem.
- When putting multiple test classes in one `<test>` in testng.xml, there is no problem.
- If specifying the listener in testng.xml instead of in `@Listeners` annotations in the test classes, there is no problem.
- If specifying the listener using `@Listeners` annotation in only one of the test classes in the suite, the listener methods are called as expected (only once) for that `<test>`.
- Specifying a different listener class for each test class, using the `@Listeners` annotation, does not help.

### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans

### Test case sample

```java
@Listeners(Listener1.class)
public class TestClass01 extends TestBase {

  @BeforeClass
  public void setup() {
    System.out.println("BeforeClass: " + this.getClass().getSimpleName());
  }

  @Test (priority = 1)
  public void test01() {
    System.out.println("inside test01()");
  }
  @Test (priority = 2)
  public void test02() {
    System.out.println("inside test02()");
  }
  @Test (priority = 3)
  public void test03() {
    System.out.println("inside test03()");
  }
}
```

testng.xml:

```xml
<?xml version = "1.0" encoding = "UTF-8"?>
<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
<suite name = "Suite01">
    <test name = "Test1">
        <classes>
            <class name = "cases.TestClass01" />
        </classes>
    </test>
    <test name = "Test2">
        <classes>
            <class name = "cases.TestClass02" />
        </classes>
    </test>
</suite>
```

Listener class:

```java
public class Listener1 implements ITestListener {

  @Override
  public void onTestStart(ITestResult result) {
    System.out.println("Listener: Running test: " + result.getName());
  }

  @Override
  public void onStart(ITestContext context) {
    System.out.println("Listener: Started: " + context.getName());
  }

  @Override
  public void onFinish(ITestContext context) {
    System.out.println("Listener: Finished: " + context.getName());
  }
}
```

TestNG console output:

```
> Listener: Started: Test1
> Listener: Started: Test1
> BeforeClass: TestClass01
> Listener: Running test: test01
> Listener: Running test: test01
> inside test01()
> Listener: Running test: test02
> Listener: Running test: test02
> inside test02()
> Listener: Running test: test03
> Listener: Running test: test03
> inside test03()
> Listener: Finished: Test1
> Listener: Finished: Test1
> Listener: Started: Test2BeforeClass: TestClass02
> Listener: Running test: test01
> inside test01()
> Listener: Running test: test02
> inside test02()
> Listener: Running test: test03
> inside test03()
> Listener: Finished: Test2
> ===============================================
> Suite01
> Total tests run: 6, Passes: 6, Failures: 0, Skips: 0
> ===============================================
```

There is a line break missing / removed in the console output, but I'm guessing it's a separate issue.

Best regards, and thanks for a great testing framework!

**Comment by QAthulhu** [comment]
@krmahadevan is the fix for this issue available now or will it be available in 7.2.0? I am currently experiencing it, also on 7.1.0

**Comment by krmahadevan** [comment]
@QAthulhu - The fix should be available only after `7.2.0` gets released. You can always cross check the updated [changes file](https://github.com/cbeust/testng/blob/master/CHANGES.txt#L27)

**Comment by QAthulhu** [comment]
@krmahadevan thanks!
